### PR TITLE
Reinitialize simulated reader upon update frequency change

### LINKED ISSUE #7973: Handle Simulated Reader Initialization on update frequency changed
context: https://github.com/woocommerce/woocommerce-android/pull/7948#discussion_r1036974002

Re-initialize the simulated card reader when the user changes the update frequency.

### PR DESCRIPTION
**Pull request by rossanafmenezes**
Closes: #7973 

### Description

Context [here](https://github.com/woocommerce/woocommerce-android/pull/7948#discussion_r1036974002)

As pointed, the code to change the update frequency on the simulated reader did not cover for one scenario which is when you connect the reader via Card Reader Manager. 

This PR adds the logic to change this behaviour and reinitializes the reader when connected via Manage Card Reader.

### Testing instructions

1. Go to Menu - Settings - Developer Options
2. Make sure the Simulated reader is enabled
3. Click on Update Simulated Row
4. Set frequency to "Always"
5. Go to Payments -> Manage card reader.
6. Start connecting to the reader.
7. Notice an update of the software starts.
8. Cancel it or wait until it completes and **disconnect from the reader**.
9. Go back to the Developer options screen and set frequency to "Never"
10. Go to Payments -> Manage card reader.
11. Start connecting to the reader.
12. Make sure the reader gets connected without an update

### Images/gif

Before:

https://user-images.githubusercontent.com/30724184/205294987-192cf122-1bf8-41fa-8f23-84f16cd000de.mp4

After: 

https://user-images.githubusercontent.com/30724184/205295013-362b9678-d039-45a9-a61c-622ec8cd51eb.mp4

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by malinajirka — APPROVED** [review]
Thanks for the changes!

> Context https://github.com/woocommerce/woocommerce-android/pull/7948#discussion_r1036974002
> As pointed, the code to change the update frequency on the simulated reader did not cover for one scenario which is when you connect the reader via Card Reader Manager.
> This PR adds the logic to change this behaviour and reinitializes the reader when connected via Manage Card Reader.

I find this description a bit confusing - I believe we don't support any other connection then through the Card Reader Manager. 

When the Card Reader Manager is initialized it also initializes the simulated terminal/reader.

However, the issue that this PR fixes is that if the Card Reader Manager is initialized and the user changes UpdateFrequency **afterwards** -> the updated frequency is never propagated to the simulated reader => this PR solves this issue by re-initializing the simulated Stripe terminal immediately after the user changes the frequency.


Either case, this PR fixes the issue :).

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/DeveloperOptionsViewModel.kt [hunk 1]
```diff
@@ -99,14 +101,26 @@ class DeveloperOptionsViewModel @Inject constructor(
...
-                DeveloperOptionsViewState.UpdateOptions.values().toList(),
+                UpdateOptions.values().toList(),
                 developerOptionsRepository.getUpdateSimulatedReaderOption()
             )
         )
     }
 
     fun onUpdateReaderOptionChanged(selectedOption: UpdateOptions) {
         developerOptionsRepository.updateSimulatedReaderOption(selectedOption)
+        if (!cardReaderManager.initialized) {
```
Comment (malinajirka) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/DeveloperOptionsViewModel.kt]: 🚫 I re-tested the app after clearing shared preferences and encountered a crash:
1. Clear app data or disable all permissions of the app (especially location permissions)
2. Open developer options
3. Change update frequency
4. 💥 

I think we need to use a slightly different approach after all - we probably don't want to initialize the card reader manager here and we want to re-initialized the reader/terminal only if it has been already initialized. If it hasn't be initialized we can only store the selected value into shared preferences and it'll be initialized with the correct value when the user attempts to connect to it. Sorry for providing misleading info last week.

**Review by malinajirka — CHANGES_REQUESTED** [review]
Found a crash ☝️

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/DeveloperOptionsViewModel.kt (hunk 1 above)
Comment (rossanafmenezes) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/DeveloperOptionsViewModel.kt]: Thank you. Fixed on [077f7b5](https://github.com/woocommerce/woocommerce-android/pull/7977/commits/077f7b501654bb28f701928c94ab2860c17c65b1)

**Review by malinajirka — APPROVED** [review]
Works as expected :shipit: Thanks!

**Pull request merged**